src/python/turicreate/toolkits/object_detector/_tf_image_augmenter.py

```diff
@@ -22,38 +22,267 @@
+    return math_ops.logical_and(math_ops.less_equal(num_attempts, max_attempts), math_ops.logical_not(perform_cropped_annotations))
+
+def _crop_augmenter_loop_perform_crop(image_box, keep_annotations, annotations, num_attempts, max_attempts, perform_cropped_annotations, aspect_ratio_var, min_aspect_ratio, max_aspect_ratio, min_height_var, max_height_var, image_height, image_width, min_area_fraction, max_area_fraction):
+    cropped_height = random_ops.random_uniform([], min_height_var, max_height_var)
+    cropped_width = cropped_height * aspect_ratio_var
+    
+    x_offset = random_ops.random_uniform([], 0.0, (image_width - cropped_width))
+    y_offset = random_ops.random_uniform([], 0.0, (image_height - cropped_height))
+        
+    unstacked_image_box = tf.unstack(image_box)
+    
+    unstacked_image_box[0] = tf.to_float(tf.to_int32(y_offset))
+    unstacked_image_box[1] = tf.to_float(tf.to_int32(x_offset))
+    unstacked_image_box[2] = tf.to_float(tf.to_int32(cropped_height))
+    unstacked_image_box[3] = tf.to_float(tf.to_int32(cropped_width))
+
+    image_box = tf.stack(unstacked_image_box)
+    
+    intersection_image_box = tf.unstack(image_box)
+    
+    intersection_image_box[0] = tf.to_float(tf.to_int32(y_offset))
+    intersection_image_box[1] = tf.to_float(tf.to_int32(x_offset))
+    intersection_image_box[2] = tf.to_float(tf.to_int32(cropped_height + y_offset))
+    intersection_image_box[3] = tf.to_float(tf.to_int32(cropped_width + x_offset))
+
+    intersection_bounding_box = tf.stack(intersection_image_box)
+    
+    identifier, box, confidence = decompose_annotations(annotations, tf.to_float(image_height), tf.to_float(image_width))
+    
+    unstacked_ann = tf.unstack(box)
+    new_annotations_arr = []
+    
+    ty = tf.to_float(tf.to_int32(y_offset))
+    tx = tf.to_float(tf.to_int32(x_offset))
+
+    # Make the transformation matrix
+    transformation = tf.reshape(tf.stack([
+        1.0,     0.0,     -ty,
+        0.0,     1.0,     -tx,
+        0.0,     0.0,    1.0]
+    ), (3, 3))
+
+    for ann in unstacked_ann:
+        intersection_points = _get_intersection_point(intersection_bounding_box, ann)
+        area_intersection = _get_area(intersection_points)
+        area_annotation = _get_area(ann)
+        area_fraction = area_intersection / area_annotation
+        
+        # TODO: Replace this condition soon
+    
+        mat_intersection_points = tf.expand_dims(intersection_points, 0)
+        transformed_intersection_points = apply_transformation(mat_intersection_points, transformation)
+        
+        overlap_condition = math_ops.greater_equal(area_fraction, 0.5)
+        new_annotations_arr.append(control_flow_ops.cond(overlap_condition,
+                                                        lambda: tf.squeeze(transformed_intersection_points),
+                                                        lambda: np.array([0.0, 0.0, 0.0, 0.0], dtype=np.float32)))
```

Comment:
For these kinds of conditionals, I usually just cast the bool to a float and multiply with it. I don't know what's faster in TF though. I tend to borrow notions from straight up compiled languages where control flow can screw up pipelining. I don't know if this notion holds in the TF graph.